### PR TITLE
core(link-elements): ignore non-HTMLLinkElements

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
updates link elements gatherer to only consider real link elements and adds a smoketest for it

**Related Issues/PRs**
fixes https://github.com/GoogleChrome/lighthouse/issues/9764

### CODE REVIEW

**Comment by brendankenny** [comment]
I was only able to test it briefly, so I could have this wrong, but from the test site https://sundaysuppermovement.com/ in #9764, it appears that the only link rel="preload" elements are SVG elements, but they are triggering preloads, so dropping these might lead to unexpected results in things like `uses-rel-preload` and etc

**Comment by patrickhulce** [comment]
>  it appears that the only link rel="preload" elements are SVG elements, but they are triggering preloads, so dropping these might lead to unexpected results in things like uses-rel-preload and etc

No they shouldn't be triggering anything, they're just invalid elements. The preload requests come from the `rel="preload" onload="this.rel = 'stylesheet'"` pattern the page uses.

**Comment by brendankenny** [comment]
> AFAICT, the preload requests come from the `rel="preload" onload="this.rel = 'stylesheet'"` pattern the page uses.

Nice catch!

Ah, sorry, I'm finally getting what you're saying. These end up as `SVGUnknownElement` (though that's not a public interface) since they're from a foreign namespace, so while they get parsed (and are apparently containers in svg, interesting) and have a `link` tagName, they are otherwise inert.

> In general, the SVG user agent must include the unknown foreign-namespaced elements in the DOM but will ignore and exclude them for rendering purposes.

This sounds good then.

File: lighthouse-cli/test/smokehouse/dbw-config.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ module.exports = {
       'render-blocking-resources',
       'errors-in-console',
       'efficient-animated-content',
+      'apple-touch-icon', // pull in apple touch icon to test `LinkElements`
```
Comment (brendankenny) [inline lighthouse-cli/test/smokehouse/dbw-config.js]: any reason to do this in dbw and not in perf or seo where LinkElements are already being used?

Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/dbw-config.js]: just that dbw felt like the dumping ground of error catching appropriate for this sort of thing 😆 

the others are so focused it felt mean to pollute them with SVG nonsense

File: types/artifacts.d.ts [hunk 2]
```diff
@@ -217,7 +217,7 @@ declare global {
         /** The `as` attribute of the link */
         as: string
         /** The `crossOrigin` attribute of the link */
-        crossOrigin: 'anonymous'|'use-credentials'|null
+        crossOrigin: string | null
```
Comment (brendankenny) [inline types/artifacts.d.ts]: this is a loss for documentation but I suppose there's always mdn :)

**Pull request merged**